pkg/applications/helmclient/client.go

```diff
@@ -0,0 +1,311 @@
+	if err != nil {
+		return nil, err
+	}
+	return rel, nil
+}
+
+// Uninstall the release in targetNamespace.
+func (h HelmClient) Uninstall(releaseName string) (*release.UninstallReleaseResponse, error) {
+	uninstallClient := action.NewUninstall(h.actionConfig)
+	return uninstallClient.Run(releaseName)
+}
+
+// buildDependencies add missing repositories and then do a Helm dependency build (ie download the chart dependencies
+// from repositories into "charts" folder).
+func (h HelmClient) buildDependencies(chartLoc string) (*chart.Chart, error) {
+	regClient, err := registry.NewClient() // todo credentials
+	if err != nil {
+		return nil, fmt.Errorf("can not initialize registry client %w", err)
+	}
+
+	var out strings.Builder
+	man := &downloader.Manager{
+		Out:              &out,
+		ChartPath:        chartLoc,
+		Getters:          h.getterProviders,
+		RepositoryConfig: h.settings.RepositoryConfig,
+		RepositoryCache:  h.settings.RepositoryCache,
+		RegistryClient:   regClient,
+		Debug:            false,
+		Verify:           downloader.VerifyNever,
+		SkipUpdate:       true,
+		// todo credentials
+	}
+
+	chartToInstall, err := loader.Load(chartLoc)
+	if err != nil {
+		return nil, fmt.Errorf("can not load char: %w", err)
+	}
+
+	fi, err := os.Stat(chartLoc)
+	if err != nil {
+		return nil, fmt.Errorf("can not find chart at `%s': %w", chartLoc, err)
+	}
+
+	// If we got the chart from the filesystem (ie cloned from a git repository). We have to do build dependencies because
+	// charts directory may not exist.
+	//
+	// note: if we got the chart from a remote helm repository, we don't have to build dependencies because the package
+	// (ie the tgz) should already contain it.
+	if fi.IsDir() {
+		// Helm does not download dependency if the repository is unknown (ie not present in repository.xml)
+		// so we explicitly add to the repository file.
+		var dependencies []*chart.Dependency
+		if chartToInstall.Lock != nil {
+			dependencies = chartToInstall.Lock.Dependencies
+		} else {
+			dependencies = chartToInstall.Metadata.Dependencies
+		}
+
+		for _, dep := range dependencies {
+			// oci or file dependencies can not be added as a repository.
+			if strings.HasPrefix(dep.Repository, "http://") || strings.HasPrefix(dep.Repository, "https://") {
+				if _, err := h.ensureRepository(dep.Repository); err != nil {
+					return nil, fmt.Errorf("can not download index for repository: %w", err)
+				}
+			}
+		}
+
+		// Equivalent of helm dependency build.
+		err = man.Build()
+		if err != nil {
+			h.logger.Errorf("can not build dependencies for chart '%s'. log='%s'", chartLoc, out.String())
+			return nil, fmt.Errorf("can not build dependencies: %w", err)
+		}
+	}
+	h.logger.Debugf("succesully build dependencies for chart '%s'. log='%s'", chartLoc, out.String())
+	return chartToInstall, nil
+}
+
+// ensureRepository add the repository url if it's not already exist and download the last index file.
```

Comment:
_adds_

_if it doesn't exist_ or _if it's not already existing_

_downloads_

"latest" might be better than "last"